internal/app/go-filecoin/retrieval_market_connector/provider.go

```diff
@@ -1,28 +1,117 @@
+	retmkt "github.com/filecoin-project/go-fil-markets/retrievalmarket"
+	rmnet "github.com/filecoin-project/go-fil-markets/retrievalmarket/network"
 	"github.com/filecoin-project/specs-actors/actors/abi"
 	"github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	xerrors "github.com/pkg/errors"
 )
 
-// RetrievalProviderNodeConnector adapts the node to provide an interface to the retrieval provider
-type RetrievalProviderNodeConnector struct{}
+// RetrievalProviderConnector is the glue between go-filecoin and retrieval market provider API
+type RetrievalProviderConnector struct {
+	bstore   blockstore.Blockstore
+	net      rmnet.RetrievalMarketNetwork
+	paychMgr PaychMgrAPI
+	provider retmkt.RetrievalProvider
+	unsealer UnsealerAPI
+}
+
+var _ retmkt.RetrievalProviderNode = &RetrievalProviderConnector{}
+
+type UnsealerAPI interface {
+	UnsealSector(ctx context.Context, sectorID uint64) (io.ReadCloser, error)
+}
 
-// NewRetrievalProviderNodeConnector creates a new connector
-func NewRetrievalProviderNodeConnector() *RetrievalProviderNodeConnector {
-	return &RetrievalProviderNodeConnector{}
+// NewRetrievalProviderConnector creates a new RetrievalProviderConnector
+func NewRetrievalProviderConnector(net rmnet.RetrievalMarketNetwork, us UnsealerAPI,
+	bs blockstore.Blockstore, paychMgr PaychMgrAPI) *RetrievalProviderConnector {
+	return &RetrievalProviderConnector{
+		bstore:   bs,
+		net:      net,
+		paychMgr: paychMgr,
+		unsealer: us,
+	}
 }
 
-// UnsealSector unseals a sector so that its pieces may be retrieved
-func (r RetrievalProviderNodeConnector) UnsealSector(ctx context.Context, sectorID uint64, offset uint64, length uint64) (io.ReadCloser, error) {
-	panic("TODO: go-fil-markets integration")
+// SetProvider sets the retrieval provider for the RetrievalProviderConnector
+func (r *RetrievalProviderConnector) SetProvider(provider retmkt.RetrievalProvider) {
+	r.provider = provider
 }
 
-// SavePaymentVoucher saves a payment voucher
-func (r RetrievalProviderNodeConnector) SavePaymentVoucher(ctx context.Context, paymentChannel address.Address, voucher *paych.SignedVoucher, proof []byte, expectedAmount abi.TokenAmount) (abi.TokenAmount, error) {
-	panic("TODO: go-fil-markets integration")
+// UnsealSector unseals the sector given by sectorId and offset with length `length`
+// It rejects offsets > int size and length > int64 size; the interface wants
+// uint64s. This would return a bufio overflow error anyway, but the check
+// is provided as a debugging convenience for the consumer of this function.
+func (r *RetrievalProviderConnector) UnsealSector(ctx context.Context, sectorID uint64,
+	offset uint64, length uint64) (io.ReadCloser, error) {
+	// reject anything that's a real uint64 rather than trying to get cute
+	// and offset that much or copy into a buf that large
+	intSz := reflect.TypeOf(0).Size()*8 - 1
+	maxOffset := uint64(1 << intSz)
```

Comment:
I'm not feeling great about using `reflect.TypeOf(0).Size()*8 - 1`. Here's a simpler maxint version I got off the internet `const MaxInt = int(^uint(0) >> 1)`.

Reply:
Thanks, your google fu is better than mine